### DISCUSSION TITLE
Can not config the chain name in langserve add_routes

### DISCUSSION THREAD

**Discussion by kientv** [langchain-ai/langserve#514] [Q&A]
```
myChain.with_config(
        run_name=chatbot_name
)
```
1. When I use langserve then chain name is `/api/v1`:
`add_routes(app, myChain, path="/api/v1",`
2. When I don't use langserve. Just call myChain.invoke than chain name is chatbot_name

I want to see chain name in langesmith of each chatbot, not same /api/v1 for every chatbot.
How to set name of chain to chatbot_name in langserve?

**Comment by eyurtsev** [comment]
This was done by design as for when debugging a web-service, it's nice to have a direct mapping with exposed endpoints.

What's your use case? Why do you need to modify it?

**Reply by kientv** [reply]
Thanks

Because I have more than one chatbot that serve in a single endpoint (difference configurable) and langserve configurable property is  chatbot_name.
